saleor/order/utils.py

```diff
@@ -1164,6 +1165,41 @@ def updates_amounts_for_order(order: Order, save: bool = True):
         )
 
 
+def updates_amounts_and_search_vector_for_order_with_lock(order: Order, update_fields):
+    with transaction.atomic():
+        # Add a transaction block to ensure that the order status won't be overridden by
+        # another process.
+        locked_order = (
+            Order.objects.prefetch_related(
+                "payments", "payment_transactions", "granted_refunds"
+            )
+            .select_for_update()
```

Comment:
I think we should unify usage of select_for_update and use here `saleor.order.utils.order_qs_select_for_update`

The problem I'm expecting is that default ordering is defined as (-number) while in a lot of places for locking purpose we use `order.id` as ordering - and this may cause some deadlocks.

Reply:
Can be ignored after consulting, we only get one Order